wolfssl/internal.h

```diff
@@ -4375,6 +4380,13 @@ extern const WOLF_EC_NIST_NAME kNistCurves[];
 #define kNistCurves_MAX_NAME_LEN 7
 #endif
 
+#if defined(OPENSSL_EXTRA) || \
+    (defined(OPENSSL_EXTRA_X509_SMALL) && !defined(NO_RSA))
+extern WC_RNG globalRNG;
```

Comment:
Why is this exposed?
Random APIs should make this unnecessary.

Reply:
I'm not sure what you mean by exposed? It is in the internal.h header so it will not be visible to users. I moved it to a header since I initialize it in wolfSSL_Init and it is easier to work with (an unused static variable generates warnings so piling on additional #define checks might not be worth it).

Reply:
@julek-wolfssl : You removed the `static` from `globalRNG` allowing extern of it from other code files. I think Sean means we should avoid using this and rely on "rng" from arguments or provided ctx structures. This "globalRNG" was added for the openssl comaptbility for the RNG API's not for general use. In fact I'd like to see this "global" rng eliminated altogether with a refactor.

Reply:
I'll see what I can do to get rid of it.

Reply:
But maybe removal of globalRNG should wait for another PR?

Reply:
A separate PR is fine and can wait. For this focus on trying to avoid using the global RNG.

Reply:
My main argument for moving it to a header file to not have to keep piling on #define's on top of the static variables to increase its scope in the file. The `globalRNG` variable is used in a number of functions already so it is no longer "local" to any single or set of functions. The list is:
```
WOLFSSL_RSA_GetRNG, line 28376:  rng = &globalRNG;
wolfSSL_RAND_Init, line 28402:  if (wc_InitRng(&globalRNG) < 0) {
wolfSSL_RAND_write_file, line 28532:  if (wc_RNG_GenerateBlock(&globalRNG, buf, bytes) != 0) {
wolfSSL_RAND_egd, line 28685:  else if (wc_RNG_DRBG_Reseed(&globalRNG, (const byte*) buf, bytes)
wolfSSL_RAND_Cleanup, line 28734:  wc_FreeRng(&globalRNG);
wolfSSL_RAND_bytes, line 28768:  rng = &globalRNG;
wolfSSL_RAND_poll, line 28802:  ret = wc_GenerateSeed(&globalRNG.seed, entropy, entropy_sz);
wolfSSL_DH_generate_key, line 29735:  rng = &globalRNG;
wolfSSL_DSA_generate_key, line 30257:  rng = &globalRNG;
wolfSSL_DSA_generate_parameters_ex, line 30358:  rng = &globalRNG;
wolfSSL_DSA_do_sign, line 30454:  rng = &globalRNG;
wolfSSL_RSA_sign_ex, line 30732:  rng = &globalRNG;
wolfSSL_EC_KEY_generate_key, line 32598:  rng = &globalRNG;
wolfSSL_ECDSA_sign, line 32780:  rng = &globalRNG;
wolfSSL_ECDSA_do_sign, line 34231:  rng = &globalRNG;
wolfSSL_EC25519_generate_key, line 42694:  rng = &globalRNG;
wolfSSL_ED25519_generate_key, line 42842:  rng = &globalRNG;
wolfSSL_EC448_generate_key, line 43028:  rng = &globalRNG;
wolfSSL_ED448_generate_key, line 43175:  rng = &globalRNG;
wolfSSL_BN_rand, line 45863:  rng = &globalRNG;
wolfSSL_BN_pseudo_rand, line 45934:  rng = &globalRNG;
wolfSSL_BN_is_prime_ex, line 46419:  rng = &globalRNG;
wolfSSL_RSA_free, line 47379:  if (rng != NULL && rng != &globalRNG) {
wolfSSL_RSA_new, line 47463:  rng = &globalRNG;
```
In my opinion it is cleaner to move it to the internal header (the user still doesn't have access to it). If you disagree then I'll move it back to `ssl.c`.